WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostPagerActivity.java

```diff
@@ -153,6 +195,226 @@ public void onPageSelected(int position) {
+                } else {
+                    // not stored locally, so request it
+                    ReaderPostActions.requestBlogPost(blogIdentifier, postIdentifier,
+                            new ReaderActions.OnRequestListener() {
+                        @Override
+                        public void onSuccess() {
+                            ReaderPost post = ReaderPostTable.getBlogPost(blogIdentifier, postIdentifier, true);
+                            ReaderEvents.PostSlugsRequestCompleted slugsResolved = (post != null) ? new ReaderEvents
+                                    .PostSlugsRequestCompleted(200, post.blogId, post.postId) : new ReaderEvents
+                                    .PostSlugsRequestCompleted(200, 0, 0);
+                            // notify that the slug resolution request has completed
+                            EventBus.getDefault().post(slugsResolved);
+                        }
+
+                        @Override
+                        public void onFailure(int statusCode) {
+                            // notify that the slug resolution request has completed
+                            EventBus.getDefault().post(new ReaderEvents.PostSlugsRequestCompleted(statusCode, 0, 0));
+                        }
+                    });
+                    mPostSlugsResolutionUnderway = true;
+                }
+            } else {
+                try {
+                    mBlogId = Long.parseLong(blogIdentifier);
+                    mPostId = Long.parseLong(postIdentifier);
+                    // set the IDs and let ReaderPostPagerActivity normally display the post
+                } catch (NumberFormatException e) {
+                    ToastUtils.showToast(this, R.string.error_generic);
+                    AppLog.e(AppLog.T.READER, e);
+                }
+            }
+        } else {
+            ToastUtils.showToast(this, R.string.error_generic);
+        }
+    }
+
+    /**
+     * Parse the URL fragment and interpret it as an operation to perform. For example, a "#comments" fragment is
+     * interpreted as a direct jump into the comments section of the post.
+     *
+     * @param uri the full URI input, including the fragment
+     */
+    private void parseFragment(Uri uri) {
+        // default to do-nothing w.r.t. comments
+        mDirectOperation = null;
+
+        if (uri == null || uri.getFragment() == null) {
+            return;
+        }
+
+        final String fragment = uri.getFragment();
+
+        final Pattern FRAGMENT_COMMENTS_PATTERN = Pattern.compile("comments", Pattern.CASE_INSENSITIVE);
+        final Pattern FRAGMENT_COMMENT_ID_PATTERN = Pattern.compile("comment-(\\d+)", Pattern.CASE_INSENSITIVE);
+        final Pattern FRAGMENT_RESPOND_PATTERN = Pattern.compile("respond", Pattern.CASE_INSENSITIVE);
+
+        // check for the general "#comments" fragment to jump to the comments section
+        Matcher commentsMatcher = FRAGMENT_COMMENTS_PATTERN.matcher(fragment);
```

Comment:
This isn't working for me. For example, this link opens in the browser instead of the app: [https://nickbradbury.com/2016/09/23/the-legend-of-nicky-poot/#comments](https://nickbradbury.com/2016/09/23/the-legend-of-nicky-poot/#comments)